### PR TITLE
Improvement/harmonize detectedobjects (solves #197)

### PR DESCRIPTION
**Pull request by CarloVanDriestenBMW**
* Harmonized proto indicies in every file #163 
* Merged #199 in this PR
* Merged MovingObjects and Vehicles
* Same structure for Detected*Moving/Stationary*Object as in DetectedTrafficSigns etc
* Removed DetectedLandmarks (cleanup)
* Removed LaneFraming
* Added limiting_structure_id to LaneBoundaries
* Removed SensorDataList

@pmai Please look into it very thoroughly! Thx!

@carsten-kuebler A review from you would also be very nice!

### CODE REVIEW

**Comment by carsten-kuebler** [comment]
@pmai Why do you index some fields starting with 1000? Is there a reason?

**Comment by pmai** [comment]
I use this approach when some fields are just sub-views/configs so that we have space to add „normal“ attribute fields to the top-level  message later on without needing to interleave them with those sub-views. Not strictly necessary but aids in keeping the fields in a somewhat readable order... Not that I think this whole field numbering business is worthy of human attention, but ... protobufs...

**Comment by carsten-kuebler** [comment]
@pmai But we don't have to keep to a monotonously increasing order, do we? We can insert fields that have a higher index. A renumbering of the indices should then take place in each major release.

**Comment by carsten-kuebler** [comment]
@CarloVanDriestenBMW I'm not happy with the merging of Vehicle with MovingObject and that message StationaryObject is also included in object.proto. From my point of view StationaryObject should be moved to landmarks.proto and object.proto renamed to movingobject.proto. 

For me the question arises then whether TrafficLight, TrafficSign and RoadMarking should also be implemented as extensions of StationaryObject (as Vehicle and MovingObject).

As a further question I asked myself whether there are other moving objects and there came to me variable advertising pillar and advertising display panels (movies as "2D moving" objects). Do we need these objects in order to simulate "2D moving objects" through animated "advertising films" on displays?

File: osi_object.proto [hunk 1]
```diff
@@ -273,31 +274,31 @@ message MovingObject
...
+            // Definition: Hatchback car with maximum length 4 m.
             //
             TYPE_SMALL_CAR = 2;
 
             // Vehicle is a medium car.
             //
-            // Definition:
+            // Definition: Hatchback or sedan with lenght between 4.5 and 5 m.
             //
             TYPE_MEDIUM_CAR = 3;
```
Comment (carsten-kuebler) [inline osi_object.proto]: should we reorder medium and compact car?

**Comment by pmai** [comment]
@carsten-kuebler yes, we don't need to use monotonically increasing field numbers, which is the other way of dealing with the interleaving; however since the interval between major releases should be fairly long (hopefully), there is a long time where you have to deal with files with non-monotonically increasing numbers, which is a bit error prone and unwieldy.

That said I don't have strong opinions one way or the other (except that we should ditch ProtoBufs in any case), feel free to renumber as wanted.

File: osi_lane.proto [hunk 2]
```diff
@@ -346,6 +317,12 @@ message LaneBoundary
     //
     optional Color color = 4;
 
+    // The ids of \c StationaryObjects which limit the corresponding lane.
+    // This field must be set if the \c LaneBoundary.type is set to
+    // # TYPE_STRUCTURE
```
Comment (AhmedMzid) [inline osi_lane.proto]: Is there any reason for the '#' before TYPE_STRUCTURE?
bildschirmfoto vom 2018-03-29 16-42-53

Comment (CarloVanDriestenBMW) [reply osi_lane.proto]: @carsten-kuebler can you check this?

**Comment by carsten-kuebler** [comment]
# is necessary for doxygen links.  But without space. 

Gesendet von IBM Verse


   AhmedMzid --- Re: [OpenSimulationInterface/open-simulation-interface] Improvement/harmonize detectedobjects (solves #197) (#200) --- 
    Von:"AhmedMzid" <notifications@github.com>An:"OpenSimulationInterface/open-simulation-interface" <open-simulation-interface@noreply.github.com>Kopie:"carsten-kuebler" <carsten.kuebler@twt-gmbh.de>, "Mention" <mention@noreply.github.com>Datum:Do. 29.03.2018 16:44Betreff:Re: [OpenSimulationInterface/open-simulation-interface] Improvement/harmonize detectedobjects (solves #197) (#200)
  
  @AhmedMzid commented on this pull request.  
 In osi_lane.proto: 
 > @@ -346,6 +317,12 @@ message LaneBoundary
     //
     optional Color color = 4;
 
+    // The ids of \c StationaryObjects which limit the corresponding lane.
+    // This field must be set if the \c LaneBoundary.type is set to
+    // # TYPE_STRUCTURE
 
 Is there any reason for the '#' before TYPE_STRUCTURE?

**Comment by pmai** [comment]
@CarloVanDriestenBMW Does the LaneBoundary/DetectedLaneBoundary and Lane/DetectedLaneBoundary split look right to you? If so, I'll rebase this and merge...

**Pull request merged**